invenio_search/engine.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015-2018 CERN.
+# Copyright (C)      2022 University Münster.
+# Copyright (C)      2022 TU Wien.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Transparency module for importing the chosen search engine.
+
+This module imports the ``elasticsearch``/``opensearchpy`` and
+``elasticsearch_dsl``/``opensearch_dsl`` packages based on availability,
+and provides them as ``search`` and ``dsl`` respectively.
+The aim of this is to make the choice of the search engine (Elasticsearch
+vs. OpenSearch) more transparent.
+"""
+
+ES = "Elasticsearch"
+OS = "OpenSearch"
+
+
+try:
```

Comment:
What is we do the following:
```
try:
    import elasticsearch
    import opensearch_dsl
except ModuleNotFoundError:

    try:
        import elasticsearch
        ....
    except ModuleNotFoundError:
        import opensearch_dsl as dsl
        ....

else:
    # it looks like both ES and OS are installed, fail
    raise ...
```
it is ugly, but it ensures that only is installed.

Reply:
indeed, with a descriptive error message that'd close the opening for the faulty case in which both dependencies are installed!